connect/runtime/src/main/java/org/apache/kafka/connect/runtime/isolation/Plugins.java

```diff
@@ -185,46 +180,105 @@ public Task newTask(Class<? extends Task> taskClass) {
         return newPlugin(taskClass);
     }
 
-    public Converter newConverter(String converterClassOrAlias) {
-        return newConverter(converterClassOrAlias, null);
+    /**
+     * Get an instance of the give class specified by the given configuration key.
+     *
+     * @param key The configuration key for the class
+     * @param t The interface the class should implement
+     * @return A instance of the class
+     */
+    private <T> T getInstance(AbstractConfig config, String key, Class<T> t) {
+        Class<?> c = config.getClass(key);
+        if (c == null)
```

Comment:
Is this actually possible? shouldn't a `CLASS` config fail to even parse if it doesn't have a valid value? (which also raises an interesting question re: parsing and ensuring the parsing context is always the right one.)

Reply:
I didn't understand it at first, but this method is ultimately called from the Worker when it sets up its own converters and when it starts up a task to instantiate and initialize the converters, etc. If the worker/connector configuration has a class that is not on the current classloader, then when the `AbstractConfig` is instantiated the value of the `Class<?> c` will be null since the class cannot be found/loaded using the current classloader. That's how we know to go looking in the plugins. Surprising indeed.

Reply:
Something still isn't right, though. Let me work on this some more.